libgadget/domain.c

```diff
@@ -317,8 +317,8 @@ static int64_t
 domain_particle_costfactor(int i)
 {
     /* We round off GravCost to integer*/
-    if(P[i].TimeBin)
-        return (1 + P[i].GravCost) * (TIMEBASE / (1 << P[i].TimeBin));
+    if(P[i].TimeBin > 0)
+        return (1 + P[i].GravCost) * (TIMEBASE / (1u << P[i].TimeBin));
```

Comment:
TimeBin also need to be cast unsigned, unfortunately...

Reply:
TimeBin has been made unsigned in the definition in partmanager.h